src/openseadragon.js

```diff
@@ -305,7 +305,7 @@
   * @property {Number} [rotationIncrement=90]
   *     The number of degrees to rotate right or left when the rotate buttons or keyboard shortcuts are activated.
   *
-  * @property {Number} [maxTilesPerFrame=1]
+  * @property {Number} [maxTilesPerFrame=10]
```

Comment:
I felt the async pipeline was penalized too much by the sycnronous rendering. Especially first load time, there are even cases that the viewport needs 100+ tiles to load at once. 1 tile per frame works nicely when browsing, but when the viewer opens, it was too blocking. Maybe we could instead bypass this option before open event?

Reply:
I have just tested it on a single duomo source. Having `maxTilesPerFrame=1` has a noticeable slowdown on the load time. Duomo loads up to level that contains 35 tiles (so around 50 in total?).

Reply:
Cool... Yeah, it's probably time for a change like this. Sounds like it's not really related to the cache overhaul... Just something you found while working on it. 

I wonder what the downsides are to making this bigger. It means there are more server transactions in-flight simultaneously, which can be slower to cancel, I think. For instance, if the user pans quickly to a different location, it might generate a lot of tile loads for the interstitial spaces, which then clog up the system for a bit, so the new location doesn't load for a while. Just trying to think things through here.

I suppose one option might be to jump a little slower, perhaps to 5 instead of 10?

Also, it's probably worth calling this out as its own item in the changelog. I know we don't usually include changelog changes in patches, but maybe this one should have them, since it's complicated enough? Either that or you should write up what the changelog should be and drop it in the description of this PR or something.

Reply:
I feel like I'm not understanding why the changes to the cache system should have an impact on how `maxTilesPerFrame` behaves. By default, the first frame should always be the lowest-res tile that fills the viewport, and it should get progressively higher resolution as new tiles become available in subsequent frames.

If this cache overhaul changes that basic functionality, I think we need to be really clear about what the new behavior is and decide if that is desirable.

Reply:
My read is that it's an unrelated change that @Aiosa is also proposing and therefore rolling into this patch. Is that correct, Aiosa? Ideally it would be a separate PR, but my suggestion to include it as a separate item in the changelog would allow it to be part of this PR if we want. That said, it's easy enough to remove from this PR so we can have a separate conversation about it in a new PR (including people who might be interested but aren't paying attention to the cache discussion).

Reply:
I did not compare the behaviour, I just noticed the issue when testing the new performance, and noticed increased value helped here. Intuitively, I think there could be a reason. Async pipeline leaves more was for the sync rendering call (more aggressive). Previously, you would have to finish getting a tile before a draw call was called. Now, you get the tile, initiate rendering, tile gets converted to a different format, another rendering, etc... Previously, you could have 1:1 render:tile load, now you could have 3:1 or worse (with plugins). Just because rendering will go ahead until it finishes, but, tile loading can take breaks several times. That's what I meant above by the penalization. There was too much unnecessary screen refreshes compared to how many tiles we loaded. Just an opinion though.

Reply:
I did not compare the behaviour, I just noticed the issue when testing the new performance, and noticed increased value helped here. Intuitively, I think there could be a reason. Async pipeline leaves more was for the sync rendering call (more aggressive). Previously, you would have to finish getting a tile before a draw call was called. Now, you get the tile, initiate rendering, tile gets converted to a different format, another rendering, etc... Previously, you could have 1:1 render:tile load, now you could have 3:1 or worse (with plugins). Just because rendering will go ahead until it finishes, but, tile loading can take breaks several times. That's what I meant above by the penalization. There was too much unnecessary screen refreshes compared to how many tiles we loaded. Just an opinion though. 

It's not necessarily true that the first tile fills the viewport. Moreover many times (due to big screens), the viewer starts at higher level zoom than the first available one. You can then observe a quick animation of sharpening image.

Reply:
That makes sense as to why bundling that change in with the cache overhaul is a good idea.

Reply:
The issue raised in https://github.com/openseadragon/openseadragon/issues/2610 is a good example of the downside to raising maxTilesPerFrame. I imagine that phenomenon would be much worse with this new value.